diff --git create_kind_cluster.sh create_kind_cluster.sh
index a15382b..5acdc16 100755
--- create_kind_cluster.sh
+++ create_kind_cluster.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git e2e_test.sh e2e_test.sh
index 5e8b4d4..76b3655 100755
--- e2e_test.sh
+++ e2e_test.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git examples/e2e-test.sh examples/e2e-test.sh
index 6157c92..f196c58 100755
--- examples/e2e-test.sh
+++ examples/e2e-test.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git examples/helloworld/k8s_deploy_test.sh examples/helloworld/k8s_deploy_test.sh
index 18ba1f6..c30c310 100755
--- examples/helloworld/k8s_deploy_test.sh
+++ examples/helloworld/k8s_deploy_test.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 # Debug
 # set -x
diff --git skylib/cmd.sh.tpl skylib/cmd.sh.tpl
index cf7bd52..b443649 100644
--- skylib/cmd.sh.tpl
+++ skylib/cmd.sh.tpl
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git skylib/k8s.bzl skylib/k8s.bzl
index 52ea705..120d1fc 100644
--- skylib/k8s.bzl
+++ skylib/k8s.bzl
@@ -29,7 +29,7 @@ def _python_runfiles(ctx, f):
     return "PYTHON_RUNFILES=${RUNFILES} %s" % _runfiles(ctx, f)
 
 def _show_impl(ctx):
-    script_content = "#!/bin/bash\nset -e\n"
+    script_content = "#!/usr/bin/env bash\nset -e\n"
 
     kustomize_outputs = []
     script_template = "{template_engine} --template={infile} --variable=NAMESPACE={namespace} --stamp_info_file={info_file}\n"
@@ -131,7 +131,6 @@ def k8s_deploy(
         visibility = None):
     """ k8s_deploy
     """
-
     if not manifests:
         manifests = native.glob(["*.yaml", "*.yaml.tpl"])
     if prefix_suffix_app_labels:
diff --git skylib/k8s_cmd.sh.tpl skylib/k8s_cmd.sh.tpl
index 83492a9..f1df283 100644
--- skylib/k8s_cmd.sh.tpl
+++ skylib/k8s_cmd.sh.tpl
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git skylib/k8s_gitops.sh.tpl skylib/k8s_gitops.sh.tpl
index 0de6b73..4521580 100644
--- skylib/k8s_gitops.sh.tpl
+++ skylib/k8s_gitops.sh.tpl
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git skylib/k8s_test_namespace.sh.tpl skylib/k8s_test_namespace.sh.tpl
index 9e98a1c..ab7cffe 100644
--- skylib/k8s_test_namespace.sh.tpl
+++ skylib/k8s_test_namespace.sh.tpl
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git skylib/kustomize/run-all.sh.tpl skylib/kustomize/run-all.sh.tpl
index 36bf318..abc5a0a 100644
--- skylib/kustomize/run-all.sh.tpl
+++ skylib/kustomize/run-all.sh.tpl
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # Copyright 2017 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git skylib/kustomize/set_namespace.sh skylib/kustomize/set_namespace.sh
index 38f4932..1218b8f 100755
--- skylib/kustomize/set_namespace.sh
+++ skylib/kustomize/set_namespace.sh
@@ -1,4 +1,4 @@
-#!/bin/bash +x
+#!/usr/bin/env bash +x
 # Copyright 2020 Adobe. All rights reserved.
 # This file is licensed to you under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License. You may obtain a copy
diff --git skylib/kustomize/tests/set_namespace_test.sh skylib/kustomize/tests/set_namespace_test.sh
index 6ccc192..abd1514 100755
--- skylib/kustomize/tests/set_namespace_test.sh
+++ skylib/kustomize/tests/set_namespace_test.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 set -euo pipefail
 
 function test_namespace_replaced {
diff --git skylib/tests/fork_join_test.sh skylib/tests/fork_join_test.sh
index 223bc27..b01e9f2 100755
--- skylib/tests/fork_join_test.sh
+++ skylib/tests/fork_join_test.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 set -euo pipefail
 
 function test_dedupe {
